VMware.CloudFoundation.CertificateManagement.psm1

```diff
@@ -1319,6 +1319,415 @@ Function Install-EsxiCertificate {
+    }
+}
+
+Function Invoke-VCFInstallCertificates {
+    Param (
+        [Parameter (Mandatory = $true)] [ValidateNotNullOrEmpty()] [String] $server,
+        [Parameter (Mandatory = $true)] [ValidateNotNullOrEmpty()] [String] $user,
+        [Parameter (Mandatory = $true)] [ValidateNotNullOrEmpty()] [String] $pass,
+        [Parameter (Mandatory = $true)] [ValidateNotNullOrEmpty()] [String] $workloadDomain
+    )
+    if (Test-VCFConnection -server $server) {
+        if (Test-VCFAuthentication -server $server -user $user -pass $pass) {
+            $domainType = Get-VCFWorkloadDomain -name $workloadDomain
+            $resourcesObject = gatherSddcInventory -domainType $domainType.type -workloadDomain $workloadDomain
+            Write-Output "Populating $($workloadDomain)-updateCertificateSpec.json"
+            $path = Join-Path $pwd ""
+            $operationTypeJson = '{
+                "operationType": "INSTALL",
+                '
+              $resourcesBodyObject += [pscustomobject]@{
+                  resources = $resourcesObject
+              }
+              $resourcesBodyObject | ConvertTo-Json -Depth 10 | Out-File -FilePath $path"temp.json"
+              Get-Content $path"temp.json" | Select-Object -Skip 1 | Set-Content $path"temp1.json"
+              $resouresJson = Get-Content $path"temp1.json" -Raw
+              Remove-Item -Path $path"temp.json"
+              Remove-Item -Path $path"temp1.json"
+              $requestCertificateSpecJson = $operationTypeJson + $resouresJson
+              $requestCertificateSpecJson | Out-File $path"$($workloadDomain)-updateCertificateSpec.json"
+
+              # Install Certificate
+              Try {
+                Write-Output "Installing Signed Certificates for Components Associated with Domain $($workloadDomain) (can take 60+ mins)"
+                $myTaskId = Set-VCFCertificate -domainName $($workloadDomain) -json $path"$($workloadDomain)-updateCertificateSpec.json"
+                $pollLoopCounter = 0
+                Do {
+                    If ($pollLoopCounter % 10 -eq 0) {
+                        Write-Output "Checking Installation of Signed Certificates for Components Associated with Domain $($workloadDomain)"
+                    }
+                    $response = Get-VCFTask $myTaskId.id
+                    If ($response.status -in "In Progress","IN_PROGRESS") {
+                        If (($pollLoopCounter % 10 -eq 0) -AND ($pollLoopCounter -gt 9)) {
+                            Write-Output "Certificate installation still in progress"
+                        }
+                        Start-Sleep 60
+                        $pollLoopCounter ++
+                    }
+                } While ($response.status -in "In Progress","IN_PROGRESS")
+                If ($response.status -eq "FAILED") {
+                    Write-Error "Workflow Completed with status $($response.status)" 
+                }
+                elseIf ($response.status -eq "SUCCESSFUL") {
```

Comment:
```suggestion
                } elseIf ($response.status -eq "SUCCESSFUL") {
```

Reply:
addressed in (f887646d2cc5df3c96a1dbb7750141c7a2d26eda)